transforms/dialog-full-screen-app-wrapper/dialog-full-screen-app-wrapper.js

```diff
@@ -0,0 +1,184 @@
+      appWrapperImport = j.importDeclaration(
+        [j.importDefaultSpecifier(j.identifier("AppWrapper"))],
+        j.stringLiteral(appWrapperImportString)
+      );
+    
+      // insert AppWrapper import after DialogFullScreen import
+      dialogFullScreenImport.insertAfter(appWrapperImport);
+
+      return true;
+    }
+  };
+
+  // filter the variable declarations against the expression declared as a child of DialogFullScreen
+  const filteredDeclarations = (variableDeclarations, expressionContainerName) => {
+    if (!variableDeclarations.length) {
+      return variableDeclarations;
+    }
+
+    const filteredDeclarations = [];
+
+    variableDeclarations.filter((vd) => {
+      const { declarations } = vd.value;
+
+      if (declarations.length && declarations[0].id.name === expressionContainerName) {
+        filteredDeclarations.push(declarations[0].init);
+      }
+    });
+
+    return filteredDeclarations;
+  }
+
+  // compute whether the given declaration already composes AppWrapper
+  const computeShouldUpdate = (declarations, appWrapperIdentifier) => {
+    let result = false;
+    declarations.forEach((declaration) => {
+      // declaration is jsx element
+      if (declaration.openingElement) {
+        result = declaration.openingElement.name.name !== appWrapperIdentifier;
+      } else if (declaration.body.openingElement) {
+        // declaration is function with no return
+        result = declaration.body.openingElement.name.name !== appWrapperIdentifier;
+      } else {
+        // declaration is function with return
+        result = declaration.body.body[0].argument.openingElement.name.name !== appWrapperIdentifier;
+      }
+    })
+
+    return result;
+  }
+
+  // adds the AppWrapper as a child of DialogFullScreen
+  const addAppWrapper = (dialogFullScreen, addedImport) => {
+    let appWrapperIdentifier = "AppWrapper";
+
+    // if there already was an import
+    if (!addedImport) {
+      const appWrapperImport = root.find(j.ImportDeclaration, {
+        source: {
+          type: "Literal",
+          value: "carbon-react/lib/components/app-wrapper",
+        },
+      });
+
+      // find identifier used for import
+      appWrapperIdentifier = appWrapperImport.get('specifiers', 0, 'local', 'name').value;
+    }
+    
+    const dialogFullScreenAppWrapper = j(dialogFullScreen).find(
+      j.JSXOpeningElement,
+      {
+        name: {
+          type: "JSXIdentifier",
+          name: appWrapperIdentifier,
+        },
+      }
+    );
+
+    // if app wrapper already exists within dialog, do nothing
+    if (dialogFullScreenAppWrapper.size()) {
+      return;
+    }
+    
+    const dialogFullScreenJSXExpressionContainer = j(dialogFullScreen).find(
+      j.JSXExpressionContainer
+    );
+      
+    let shouldUpdate = false;
+
+    // if there are no expression containers or multiple instances then add wrapper
+    if (dialogFullScreenJSXExpressionContainer.size() !== 1) {
```

Comment:
What if there are multiple instances but one of them is a direct child of `DialogFullScreen`?

Reply:
I think this would add a second `AppWrapper` if I'm reading this all correctly

Reply:
In that scenario it would return 1 and transform if I'm understanding you right

Reply:
I don't think so, although what I'm talking about is a very niche scenario so let's just leave it

Reply:
apologies it should say `wouldn't` return 1 for the size